raw/fetch.py

```diff
@@ -0,0 +1,239 @@
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import configparser
+import datetime
+import json
+import os
+import re
+import subprocess
+from google.cloud import secretmanager
+from typing import Tuple
+
+DIRECTORY = "/weather/config_files"
+FIELD_NAME = "date"
+PROJECT = os.environ.get("PROJECT")
+REGION = os.environ.get("REGION")
+BUCKET = os.environ.get("BUCKET")
+SDK_CONTAINER_IMAGE = os.environ.get("SDK_CONTAINER_IMAGE")
+MANIFEST_LOCATION = os.environ.get("MANIFEST_LOCATION")
+API_KEY_PATTERN = re.compile(r'^API_KEY_\d+$')
+API_KEY_LIST = []
+
+
+def new_config_file(config_file: str, field_name: str, additional_content: str,
+    co_file: bool, single_level_file: bool, first_day_first_prev: datetime.date,
+    last_day_first_prev: datetime.date, first_day_third_prev: datetime.date,
+    last_day_third_prev: datetime.date, sl_year: str, sl_month: str,
+    sl_first_date: str, sl_last_date: str) -> None:
+    """Modify the specified configuration file with new values.
+
+    Parameters:
+        config_file (str): The path to the configuration file to be modified.
+        field_name (str): The name of the field to be updated with the new value.
+        additional_content (str): The additional content to be added under the 
+                                    '[selection]' section.
+        co_file (bool): True if the configuration file is a 'CO' type file, False 
+                        otherwise.
+        single_level_file (bool): True if the configuration file contains 'sfc' in 
+                                filename, False otherwise.
+        first_day_first_prev (datetime.date): The first day of the first previous month.
+        last_day_first_prev (datetime.date): The last day of the first previous month.
+        first_day_third_prev (datetime.date): The first day of the third previous month.
+        last_day_third_prev (datetime.date): The last day of the third previous month.
+        sl_year (str): The year of the third previous month in 'YYYY' format.
+        sl_month (str): The month of the third previous month in 'MM' format.
+        sl_first_date (str): The first date of the third previous month in 'DD' format.
+        sl_last_date (str): The last date of the third previous month in 'DD' format.
+    """
+
+    config = configparser.ConfigParser()
+    config.read(config_file)
+
+    if single_level_file:
+        config.set("selection", "year", sl_year)
+        config.set("selection", "month", sl_month)
+        config.set("selection", "day", f"{sl_first_date}/to/{sl_last_date}")
```

Comment:
Will this always be the full month? If so, the value for `day` can just be `"all"`.

Reply:
Yes this always be the full month. I changed the value for `day` to `all` instead of the giving this to range and all affected code also be changed.